Huge Ultraviolet Aardvark

high

# Feemanage.withdraw function won't work for most tokens when withdrawing trapped assets

## Summary
Feemanage.withdraw function won't work for most tokens when withdrawing trapped assets because it uses safetransferFrom to transfer its own balance rather than safetransfer and there is no function to approve its address.
## Vulnerability Detail
The current implementetion of withdraw function used to withdraw trapped asset  will fail for most tokens. 
```solidity
 function withdraw(address asset_, uint256 amount_, address to_)
        external
        onlyRolesOrOwner(ADMIN_ROLE)
    {
        _transfer(asset_, amount_, address(this), to_);
    }
```
The internal _transfer function will try to use safetransferFrom to transfer its own balance to `to` but as it does not have allowance for its own address will always revert and there is not function to do the approval 
```solidity
function _transfer(address asset_, uint256 amount_, address from_, address to_) internal {
        if (asset_ == ETH_ADDRESS) {
            to_.safeTransferETH(amount_);
        } else {
            asset_.safeTransferFrom(from_, to_, amount_);
        }
    }
```
## Impact
This could lead to permanent lost of the funds
## Code Snippet
https://github.com/sherlock-audit/2024-04-titles/blob/main/wallflower-contract-v2/src/fees/FeeManager.sol#L352

https://github.com/sherlock-audit/2024-04-titles/blob/main/wallflower-contract-v2/src/fees/FeeManager.sol#L461
## Tool used

Manual Review

## Recommendation
Since the _transfer is used in so many other places i suggest you implement the transfers directly on the withdraw function